kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/PaginatedScanSuite.scala

```diff
@@ -237,62 +325,59 @@ class PaginatedScanSuite extends AnyFunSuite with TestUtilsWithTableManagerAPIs
       pageSize = 4,
-      expFileCnt = 4,
-      expBatchCnt = 1,
-      expLogFile = JSON_FILE_2,
-      expRowIdx = 4),
+      expFirstPageFileCnt = 4,
+      expFirstPageBatchCnt = 1,
+      expLastReadLogFileInFirstPage = JSON_FILE_2,
+      expLastReadRowIdxInFirstPage = 4),
     // Kernel is asked to read the 1st page of size 5. Kernel reads batch 1 and 2 in JSON_FILE_2,
     // so returns 5 AddFiles and ends at the 6th row (index 5) in JSON_FILE_2.
     SinglePageRequestTestCase(
       pageSize = 5,
-      expFileCnt = 5,
-      expBatchCnt = 2,
-      expLogFile = JSON_FILE_2,
-      expRowIdx = 5),
+      expFirstPageFileCnt = 5,
+      expFirstPageBatchCnt = 2,
+      expLastReadLogFileInFirstPage = JSON_FILE_2,
+      expLastReadRowIdxInFirstPage = 5),
     // Kernel is asked to read the 1st page of size 7. Kernel reads all batches in JSON_FILE_2,
     // batch 1 in JSON_FILE_1, so returns 9 AddFiles and ends at the 5th row (index 4)
     // in JSON_FILE_1.
     // Note: Kernel should return full batches, so return 3 full batches (and go over page limit).
     SinglePageRequestTestCase(
       pageSize = 7,
-      expFileCnt = 9,
-      expBatchCnt = 3,
-      expLogFile = JSON_FILE_1,
-      expRowIdx = 4),
+      expFirstPageFileCnt = 9,
+      expFirstPageBatchCnt = 3,
+      expLastReadLogFileInFirstPage = JSON_FILE_1,
+      expLastReadRowIdxInFirstPage = 4),
     // Kernel is asked to read the 1st page of size 9. Kernel reads all batches in JSON_FILE_2,
     // batch 1 in JSON_FILE_1, so returns 9 AddFiles and ends at the 5th row (index 4)
     // in JSON_FILE_1.
     SinglePageRequestTestCase(
       pageSize = 9,
-      expFileCnt = 9,
-      expBatchCnt = 3,
-      expLogFile = JSON_FILE_1,
-      expRowIdx = 4),
+      expFirstPageFileCnt = 9,
+      expFirstPageBatchCnt = 3,
+      expLastReadLogFileInFirstPage = JSON_FILE_1,
+      expLastReadRowIdxInFirstPage = 4),
     // Kernel is asked to read the 1st page of size 18. Kernel reads all batches in JSON_FILE_2,
     // JSON_FILE_1 and SON_FILE_0, so returns 15 AddFiles and ends at the last row (index 7)
     // in JSON_FILE_0.
     // Note: page size won't be reached because there are 15 data files in total.
     SinglePageRequestTestCase(
       pageSize = 18,
-      expFileCnt = 15,
-      expBatchCnt = 6,
-      expLogFile = JSON_FILE_0,
-      expRowIdx = 7)).foreach { testCase =>
+      expFirstPageFileCnt = 15,
+      expFirstPageBatchCnt = 6,
+      expLastReadLogFileInFirstPage = JSON_FILE_0,
+      expLastReadRowIdxInFirstPage = 7)).foreach { testCase =>
     test(s"Multiple JSON files - page size ${testCase.pageSize}") {
-      val tablePath = getTestResourceFilePath("kernel-pagination-all-jsons")
-      val nextPageToken = runSinglePaginationTestCase(
-        testName = s"Multiple JSON files - page size ${testCase.pageSize}",
-        tablePath = tablePath,
-        tableVersionOpt = Optional.empty(),
-        testCase)
+      runCompletePaginationTest(
```

Comment:
so this looks like you are refactoring existing tests now? 

could we split off this prefactor into its own PR? that would help make this PR (>950 LOC) a bit smaller, wdyt?